classes/query/QueryParticipant.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * @file classes/query/QueryParticipant.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class QueryParticipant
+ *
+ * @brief Class for QueryParticipant.
+ */
+
+namespace PKP\query;
+
+use APP\facades\Repo;
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+class QueryParticipant extends Model
+{
+    use HasCamelCasing;
+
+    protected $table = 'query_participants';
+    protected $primaryKey = 'query_participant_id';
+    public $timestamps = false;
+
+    protected $fillable = [
+        'queryId', 'userId'
+    ];
+
+    protected function casts(): array
+    {
+        return [
+            'queryId' => 'int',
+            'userId' => 'int'
+        ];
+    }
+    public function toQuery(): BelongsTo
+    {
+        return $this->belongsTo(Query::class, 'query_id', 'query_id');
+    }
+
+    /**
+     * Accessor and Mutator for primary key => id
+     */
+    protected function id(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value, $attributes) => $attributes[$this->primaryKey] ?? null,
+            set: fn($value) => [$this->primaryKey => $value],
+        );
+    }
```

Comment:
here the accessor for id may not be necessary as eloquent handles the primary key automatically. since it uses `$primaryKey` property to manage the primary key. so `$queryParticipant->id` should work without any issue

Reply:
I think this is needed so we can still use the `id` attribute since our primary key has a custom name (e.g. `query_id` in this case, rather than the default `id`).